### PR TITLE
HV-1245 BVAL-562 follow-up and include type parameter information in path

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1245

Requires https://github.com/beanvalidation/beanvalidation-api/pull/90 .

Note that I'm not convinced by the string representation. I would admit that I would have a tendency to favor consistency over readability here but it seems like a good step forward anyway.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@gsmet seems I cannot push to this PR.

**Comment by gsmet** [comment]
@gunnarmorling it's weird, the `Allow edits from maintainers.` checkbox is checked. I unchecked it then checked it again.

**Comment by gunnarmorling** [comment]
Ok, still doesn't work. I'll keep my changes locally then for now.

File: engine/src/test/java/org/hibernate/validator/test/internal/engine/typeannotationconstraint/TypeAnnotationConstraintTest.java [hunk 1]
```diff
@@ -167,17 +165,11 @@ public void property_path_contains_index_information_for_list() {
...
-		Path.Node firstNode = propertyPathIterator.next();
-		assertThat( firstNode.getIndex() ).isNull();
-		assertThat( firstNode.getName() ).isEqualTo( "names" );
-		assertThat( firstNode.getKind() ).isEqualTo( ElementKind.PROPERTY );
-
-		Path.Node secondNode = propertyPathIterator.next();
-		assertThat( secondNode.getIndex() ).isEqualTo( 0 );
-		assertThat( secondNode.getName() ).isEqualTo( NodeImpl.ITERABLE_ELEMENT_NODE_NAME );
-		assertThat( secondNode.getKind() ).isEqualTo( ElementKind.PROPERTY );
+		assertThat( constraintViolations ).containsOnlyPaths(
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/validator/test/internal/engine/typeannotationconstraint/TypeAnnotationConstraintTest.java]: Wow, seems like quite a lot of work to re-write all these :)

**Pull request closed without merging**